kernel/src/platform/platform.rs

```diff
@@ -0,0 +1,171 @@
+    /// platform will use to route syscalls.
+    fn system_call_dispatch(&self) -> &Self::SystemCallDispatch;
+
+    /// Returns a reference to the implementation of the SystemCallFilter this
+    /// platform wants the kernel to use.
+    fn system_call_filter(&self) -> &Self::SystemCallFilter;
+
+    /// Returns a reference to the implementation of the ProcessFault handler
+    /// this platform wants the kernel to use.
+    fn process_fault(&self) -> &Self::ProcessFault;
+
+    /// Returns a reference to the implementation of the Scheduler this platform
+    /// wants the kernel to use.
+    fn scheduler(&self) -> &Self::Scheduler;
+
+    /// Returns a reference to the implementation of the SchedulerTimer timer
+    /// for this platform.
+    fn scheduler_timer(&self) -> &Self::SchedulerTimer;
+
+    /// Returns a reference to the implementation of the WatchDog on this
+    /// platform.
+    fn watchdog(&self) -> &Self::WatchDog;
+}
+
+/// Configure the system call dispatch mapping.
+///
+/// Each board should define a struct which implements this trait. This trait is
+/// the core for how syscall dispatching is handled, and the implementation is
+/// responsible for dispatching to drivers for each system call number.
+///
+/// ## Example
+///
+/// ```ignore
+/// struct Hail {
+///     console: &'static capsules::console::Console<'static>,
+///     ipc: kernel::ipc::IPC,
+///     dac: &'static capsules::dac::Dac<'static>,
+/// }
+///
+/// impl SystemCallDispatch for Hail {
+///     fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
+///     where
+///         F: FnOnce(Option<&dyn kernel::SyscallDriver>) -> R,
+///     {
+///         match driver_num {
+///             capsules::console::DRIVER_NUM => f(Some(self.console)),
+///             kernel::ipc::DRIVER_NUM => f(Some(&self.ipc)),
+///             capsules::dac::DRIVER_NUM => f(Some(self.dac)),
+///
+///             _ => f(None),
+///         }
+///     }
+/// }
+/// ```
+pub trait SystemCallDispatch {
+    /// Platform-specific mapping of syscall numbers to objects that implement
+    /// the Driver methods for that syscall.
+    ///
+    ///
+    /// An implementation
+    fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
+    where
+        F: FnOnce(Option<&dyn SyscallDriver>) -> R;
+}
+
+/// Trait for implementing system call filters that the kernel uses to decide
+/// whether to handle a specific system call or not.
+pub trait SystemCallFilter {
+    /// Check the platform-provided system call filter for all non-yield system
+    /// calls. If the system call is allowed for the provided process then
+    /// return `Ok(())`. Otherwise, return `Err()` with an `ErrorCode` that will
+    /// be returned to the calling application. The default implementation
+    /// allows all system calls.
+    ///
+    /// This API should be considered unstable, and is likely to change in the
+    /// future.
+    fn filter_syscall(
```

Comment:
```suggestion
    fn filter_system_call(
```

Several functions seem to use `...system_call...`

Reply:
There are?

```
$ git grep system_call | wc -l
       0
$ git grep syscall | wc -l
     560
```

Reply:
I was referring top the internal kernel functions like:
  - platform.rs:43 
  - platform.rs:47

Reply:
Haha ok I guess I should have greped on my own PR. I don't feel compelled to change our naming convention, since we exclusively use "syscall" so far, and therefore added a patch which removes `SystemCall` and `system_call_`.